res/moves/meson.build

```diff
@@ -0,0 +1,19 @@
+pl_waza_tbl_subdirs = run_command('ls', '-d', '*/').stdout().splitlines()
```

Comment:
As a general rule, source files and directories should be explicitly listed. Globbing/listing them through an external tool is hack-ish, as Meson will not pick up new files when they are added, Refer to: https://mesonbuild.com/FAQ.html#why-cant-i-specify-target-files-with-a-wildcard

Reply:
As I've outlined previously, assuming that we do that, this means that there are two options:

1. We pass all of the input sources through the arg-list. This is, frankly, untenable; there are almost 500 moves here.
2. We create each of the binary files individually and stitch them together later. This isn't unreasonable, but it means that adding a new move is a more cumbersome process for an end-user, because now they have to interface with the build-system to make sure their files are included.

Reply:
> As I've outlined previously, assuming that we do that, this means that there are two options:
> 
> 
> 
> 1. We pass all of the input sources through the arg-list. This is, frankly, untenable; there are almost 500 moves here.
> 

I'm not sure I explained myself correctly. You would just have to list the move subdirectories explicitly instead of relying on `ls`. It's totally fine to then iterate over that list to build source file paths, like the rest of your code is already doing.

> 2. This isn't unreasonable, but it means that adding a new move is a more cumbersome process for an end-user, because now they have to interface with the build-system to make sure their files are included. Indeed, I can foresee many a query pertaining to "I made a new move, why isn't it showing up?" whenever people start trying to use this for hacks because the fail-case in this scenario is silent.

That's gonna happen even as things are now. Meson won't regenerate the build system just because you added a new move. You would have to manually wipe your build tree and reconfigure directly. I personally believe having users add an entry into a list in a meson.build file isn't the end of the world. It's better than having to perform a reconfigure.